### ISSUE TITLE
Help with improving BLEU for en-fr Supervised model

### ISSUE THREAD

**Issue by bhardwaj1230** [facebookresearch/XLM#122]
https://github.com/facebookresearch/XLM/issues/2#issuecomment-504503433
Replying to: https://github.com/facebookresearch/XLM/issues/2#issuecomment-503741941

Hello @glample , @yilinyang7 ,

I got the issue above, my pre-processing wasn't proper, I have done this time properly and got the below accuracy :

INFO - 07/01/19 18:48:54 - 14:55:16 - epoch -> 62.000000
INFO - 07/01/19 18:48:54 - 14:55:16 - valid_en-fr_mt_ppl -> 8.588624
INFO - 07/01/19 18:48:54 - 14:55:16 - valid_en-fr_mt_acc -> 66.097526
INFO - 07/01/19 18:48:54 - 14:55:16 - valid_en-fr_mt_bleu -> 30.170000
INFO - 07/01/19 18:48:54 - 14:55:16 - test_en-fr_mt_ppl -> 8.628446
INFO - 07/01/19 18:48:54 - 14:55:16 - test_en-fr_mt_acc -> 67.037034
INFO - 07/01/19 18:48:54 - 14:55:16 - test_en-fr_mt_bleu -> 31.570000

This model I ran is on a Canadian official en-fr parallel data with approximately 30% noise in the data, and test data is taken randomly from the similar data-set (Canadian official). 
- Please let me know if the results are good ?
- How can I improve this ?
- Do I need to train a language model - TLM and use this pre-training to improve this results ?

PS: Also attached the complete log file in this.

[log.txt](https://github.com/facebookresearch/XLM/files/3379433/log.txt)

command:

command: python train.py --exp_name bt_test --dump_path './dumped/testing/' --data_path './data/processed/en-fr/' --lgs 'en-fr' --n_layers 6 --n_heads 8 --dropout '0.1' --attention_dropout '0.1' --gelu_activation true --batch_size 32 --emb_dim 1024 --bptt 256 --optimizer 'adam_inverse_sqrt,beta1=0.9,beta2=0.98,lr=0.0001' --epoch_size 50000 --eval_bleu True --mt_steps 'en-fr' --encoder_only False --validation_metrics 'valid_en-fr_mt_bleu' --stopping_criterion 'valid_en-fr_mt_bleu,10' --tokens_per_batch 2000
                                     
INFO - 07/01/19 03:55:42 - 0:02:04 - ============ Data summary
INFO - 07/01/19 03:55:42 - 0:02:04 - Parallel data      - train -        en-fr:  58743714
INFO - 07/01/19 03:55:42 - 0:02:04 - Parallel data      - valid -        en-fr:      4973
INFO - 07/01/19 03:55:42 - 0:02:04 - Parallel data      -  test -        en-fr:      4961

Thank you.
Shivendra

**Comment by glample** [comment]
30 BLEU is usually a good BLEU score, but it really depends on the data set you consider. On WMT'14 and newstest 2014 SOTA is above 40 BLEU (with the code of this repo you should be able to get 41.0 with a simple greedy decoding), but on other datasets it's hard to get such a good BLEU. You need to compare with existing papers to have an idea of how good this is.

TLM will probably not help here given the amount of parallel data you have (58M parallel sentences is a lot, and pretraining is usually helpful for language pairs with small amount of parallel data).
